### PR TITLE
Change test_repositories setUp method to clear the REPOSITORIES caching

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/mozilla_ci_tools/452)

### CODE REVIEW

File: test/test_zbuildbot_bridge.py [hunk 1]
```diff
@@ -1,21 +1,34 @@
 """This file contains tests for mozci/sources/buildbot_bridge.py."""
```
Comment (armenzg) [inline test/test_zbuildbot_bridge.py]: Why do you rename the test?

File: setup.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
         'requests>=2.8.1',
         'taskcluster>=0.0.28',
         'treeherder-client>=2.0.1',
+        'jsonschema>=2.5.1'
```
Comment (armenzg) [inline setup.py]: This is from the other PR.

File: mozci/sources/buildbot_bridge.py [hunk 3]
```diff
@@ -77,17 +77,14 @@ def _create_task(buildername, repo_name, revision, metadata=None, task_graph_id=
     """
     if not valid_builder(buildername):
         raise MozciError("The builder '%s' is not a valid one." % buildername)
-
     builder_info = get_buildername_metadata(buildername)
     if builder_info['repo_name'] != repo_name:
         raise MozciError(
             "The builder '%s' should be for repo: %s." % (buildername, repo_name)
         )
-
```
Comment (armenzg) [inline mozci/sources/buildbot_bridge.py:86]: These white lines are intentional for visual purposes.

File: test/test_zbuildbot_bridge.py [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
-            builders=self.buildernames,
-            revision=self.revision,
-            complete=False
-        )
+        self.repo_name = 'repo'
+        self.builders_graph = {self.buildernames[0]: None}
 
-    def test_task_name_metadata_is_buildername(self):
+    @patch('mozci.sources.buildbot_bridge.valid_builder', return_value=True)
+    def test_task_name_metadata_is_buildername(self, mock1):
```
Comment (armenzg) [inline test/test_zbuildbot_bridge.py]: It seems that Travis now complains about this.
If you want to mock the fetching of allthethings.json you can look at test_allthethings.py to see how to do it.

**Comment by martiansideofthemoon** [comment]
@armenzg r?

**Comment by martiansideofthemoon** [comment]
@armenzg it was a one line fix! My test was adding repositories data to the in memory cache. I just needed to empty it.
@armenzg could we merge it? I'd love to see `build: passing` on the README :smile:

**Comment by armenzg** [comment]
This test? https://github.com/mozilla/mozilla_ci_tools/commit/288b525ed5b6f97cf40f004a6837e1c6a2cbdd87

I'm glad you caught this! Very odd.

**Pull request merged**